src/Peachpie.CodeAnalysis/Utilities/DistinctQueue.cs

```diff
@@ -5,18 +5,27 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using Pchp.CodeAnalysis.Semantics.Graph;
+using Peachpie.CodeAnalysis.Utilities;
 
 namespace Pchp.CodeAnalysis.Utilities
 {
     /// <summary>
-    /// Represents queue where items are enqueued just once and queue can be accessed in parralel.
+    /// Represents priority queue where items are enqueued just once and queue can be accessed in parallel.
     /// </summary>
-    internal sealed class DistinctQueue<T>
+    internal sealed class DistinctQueue<T> where T : BoundBlock
     {
         readonly object _syncRoot = new object();
 
-        readonly HashSet<T> _set = new HashSet<T>();
```

Comment:
`IsEnqueued` does not seems right - now we cannot use the single instance of T within more than one `DistinctQueue` and `DistinctQueue` would require its items to always have `IsEnqueued` property. Also the class `DistinctQueue` is now unusable for general use.

Reply:
OK, I've probably over-optimised this bit O:-) It's reverted back to the usage of ```HashSet```.